packages/orbit-components/src/InputGroup/index.tsx

```diff
@@ -152,7 +151,8 @@ const InputGroup = React.forwardRef<HTMLFieldSetElement, Props>(
                   <div
                     key={randomId(String(key))}
                     className={cx(
-                      "orbit-input-group-child pe-200 last:p-0 [&_.orbit-input-field-fake-input]:hidden [&_.orbit-input-field-fake-input]:bg-transparent [&_.orbit-input-field-input~.orbit-input-field-fake-input]:shadow-none [&_.orbit-select-container_select]:bg-transparent [&_.orbit-select-container_select]:shadow-none [&_.orbit-select-container_select]:focus:outline-none",
+                      "orbit-input-group-child [&_.orbit-input-field-fake-input]:hidden [&_.orbit-input-field-fake-input]:bg-transparent [&_.orbit-input-field-input~.orbit-input-field-fake-input]:shadow-none [&_.orbit-select-container_select]:bg-transparent [&_.orbit-select-container_select]:shadow-none [&_.orbit-select-container_select]:focus:outline-none",
+                      !childFlex && "w-full",
```

Comment:
Is this change intentional?

e.g. if I remove the inline `flex: 1 1 100%` from this Storybook story, then I get this "broken" state wrt to focus outline:
https://kiwicom-orbit-domi-inputgroup-a11y.surge.sh/?path=/story/inputgroup--date-of-birth&globals=backgrounds.grid:!false;viewport:largeDesktop

![Screenshot 2025-03-25 at 11 26 42](https://github.com/user-attachments/assets/80990cfd-8f97-43e8-9f66-b275c68450cb)

Reply:
Not sure if it's the cause or even related, it just seemed like you had some consideration for when the child is a flex container or not, so it could be.

Reply:
Thanks for checking! Yes, I was trying to fix the focus indicator but I missed this scenario 🙈 

I did the changes so it behaves like this now:

- If flex value is provided for all children, it's applied for all
- If flex is provided for less than number of children (case from your screenshot), the corresponding values are applied, then first flex value is set for the remaining children (this is current implementation from production) and then the last child takes up the rest of space
- if only one flex value is provided (e.g. string or array with one value), it's set for all except the last child - last one takes up the rest

WDYT?

Reply:
There is this comment in the code:

> // TODO: next cleanup iteration just remove this whole `flex` prop thing
> // and cloning elements, and make children take care of their sizing themselves

It just doesn't seem to be many use cases in our codebases for this prop, so I would even propose to remove that logic and migrate the necessary parts to be explicit about the flex behaviour. Let's discuss with the team.

(e.g. we could add in Playroom some specific code snippets for given components with reference implementations)

Reply:
Yes, I saw this one but I wasn't sure if this is something we want to do already as part of this task.

I'll create new thread in the channel to discuss this then.